mix.exs

```diff
@@ -44,8 +44,8 @@ defmodule AWS.Mixfile do
       {:earmark, "~> 1.4", only: [:dev]},
       {:ex_doc, "~> 0.21", only: [:dev]},
       {:bypass, "~> 2.1", only: [:test]},
-      {:httpoison, "~> 1.6"},
-      {:poison, "~> 4.0"}
+      {:jason, "~> 1.2"},
+      {:hackney, "~> 1.16"}
```

Comment:
While at it, I think we should do this:

```suggestion
      {:hackney, "~> 1.16", optional: true}
```

and then wherever we use `:hackney` we guard it:

```
if Code.ensure_loaded?(:hackney) do
  :hackney.request(...)
else
  Logger.error(...)
  raise "missing hackney dependency"
end
```

We could have an error message similar to https://github.com/swoosh/swoosh/blob/v1.1.2/lib/swoosh/api_client.ex#L64:L78.

All of it is so that if a project happens to be already using a different http client, they won't unnecessarily download hackney.

In theory we could do it for the json library too, but because Jason is much smaller dependency I think it's not worth it as the vast majority of users would be using it anyway.

Reply:
Agreed! I will update the PR soon! Thanks 😄

Reply:
Thanks for waiting. It's ready for review :D